### PR TITLE
Homework3: MarinaParomenskaya

### PR DESCRIPTION
**Pull request by MarinaParomenskaya**

### CODE REVIEW

File: Homework3/Codewars.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# The function should return true if the string is valid,
+# and false if it's invalid.
+# "()"              =>  true
+# ")(()))"          =>  false
+# "("               =>  false
+# "(())((()())())"  =>  true
+
+string = '(())((()())())'
+count = 0
+for char in string:
```
Comment (belovmd) [inline Homework3/Codewars.py:13]: more interesting task - we have 3 types of parentheses [{()}]

File: Homework3/Fizz_buzz.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+line_count = 0
+for i in range(1, 101):
+    current = ''
+    if i % 3 == 0:
+        current = 'Fizz'
+    if i % 5 == 0:
+        current += 'Buzz'
+    if current == '':
+        current = str(i)
```
Comment (belovmd) [inline Homework3/Fizz_buzz.py]: we don't need str if we want just to print it

File: Homework3/Fizz_buzz.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    print(current, end=' ')
+    line_count += 1
+    if line_count == 20:
+        line_count = 0
+        print('\t')
```
Comment (belovmd) [inline Homework3/Fizz_buzz.py:20]: \n ?
you print just tabulation

File: Homework3/List_practice.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# ['ab', 'ac', 'ad', 'bb', 'bc', 'bd'].
+lst1 = [ch1 + ch2 for ch1 in 'ab' for ch2 in 'bcd']
+print(lst1)
+# 2. Используйте на предыдущий список slice чтобы получить
+# следующий: ['ab', 'ad', 'bc'].
+lst2 = lst1[::2]
+print(lst2)
+# 3. Используйте генератор списков чтобы получить следующий
+# ['1a', '2a', '3a', '4a'].
+lst3 = [str(num1 + 1) + 'a' for num1 in range(4)]
```
Comment (belovmd) [inline Homework3/List_practice.py]: '1234' ?

**Review by belovmd — COMMENTED** [review]
1. Please remove second HW tasks
2. Please address these issues

File: Homework3/List_practice.py (hunk 4 above)
Comment (MarinaParomenskaya) [reply Homework3/List_practice.py]: Of course! ))

File: Homework3/Fizz_buzz.py (hunk 2 above)
Comment (MarinaParomenskaya) [reply Homework3/Fizz_buzz.py]: ok

File: Homework3/Fizz_buzz.py (hunk 3 above)
Comment (MarinaParomenskaya) [reply Homework3/Fizz_buzz.py:20]: yes. becouse '\n' makes extra space between rows

File: Homework3/Codewars.py (hunk 1 above)
Comment (MarinaParomenskaya) [reply Homework3/Codewars.py:13]: done

File: Homework3/Codewars.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+string = '(({)){}((}()[[]()])())'
+parentheses = list('(){}[]')
+dct = {}
+for char in string:
+    if char in parentheses:
+        index = parentheses.index(char)
+        ch = parentheses[index - 1] if index % 2 else char
+        dct[ch] = dct.get(ch, 0) + (1 if not index % 2 else -1)
+print('True' if sum(el for el in dct.values()) == 0 else 'False')
```
Comment (belovmd) [inline Homework3/Codewars.py:18]: this code returns True for your string, but it's false: {)) - is incorrect sequence

**Review by belovmd — COMMENTED** [review]
9

**Comment by MarinaParomenskaya** [comment]
Добрый день!

Не совсем поняла, почему последовательность "{))" в моем примере
некорректна.
Поясните, пожалуйста.

пн, 17 февр. 2020 г. в 16:47, Maxim Belov <notifications@github.com>:

**Pull request closed without merging**